smt/exprs.cpp

```diff
@@ -117,7 +137,7 @@ template<> DisjointExpr<expr>::DisjointExpr(const expr &e, bool unpack_ite,
           }
 
           for (auto &[rhs_v, rhs_domain] : rhs) {
-            add(lhs_v.concat(rhs_v.subst(from, to).simplify()),
+            add(lhs_v.concat(subst_for_ptr(rhs_v, from, to).simplify()),
```

Comment:
is this semantically equivalent as the previous code?
can you do an `assert(rhs_v.subst(from, to).eq(subst_for_ptr(rhs_v, from, to)))` and run over the tests to see if it fails?

Reply:
No problem. I couldn't run it yet (had a few things to do), but will be able to run on single file benchmarks after then.